djitellopy/tello.py

```diff
@@ -1057,12 +1060,39 @@ def update_frame(self):
         """
         try:
             for frame in self.container.decode(video=0):
-                self.frame = np.array(frame.to_image())
+                with self.lock:
+                    self.frames.put(np.array(frame.to_image()))
+
                 if self.stopped:
                     self.container.close()
                     break
         except av.error.ExitError:
             raise TelloException('Do not have enough frames for decoding, please try again or increase video fps before get_frame_read()')
+    
+    def get_queued_frame(self):
+        """
+        Get a frame from the queue
+        """
+        with self.lock:
+            try:
+                return self.frames.get(block=False)
+            except Empty:
+                return None
+
+    def get_latest_frame(self):
+        """
+        Get the latest frame from the queue
+        """
+        with self.lock:
+            if self.frames.not_empty():
+                return self.frames.queue[-1]
+        
+    def get_frame(self):
+        """
+        Access the frame variable directly
+        """
+        with self.lock:
+            return self.frame
```

Comment:
are you even setting `self.frame` anywhere?

I agree, simply accessing the `BackgroundFrameRead.frame` from outside the class is probably a bad idea.
For keeping compatability we could use `@property` for providing the old `.frame` behaviour, but making it basically call `self.get_latest_frame`